R/n_subject.R

```diff
@@ -93,40 +95,70 @@ avg_event <- function(id, group, par = NULL) {
     db <- data.frame(id = id, group = group)
     res <- table(db$id, db$group)
     res <- data.frame(res)
-    avg <- vapply(split(res, res$Var2), function(x) mean(x$Freq, na.rm = TRUE), FUN.VALUE = numeric(1))
-    se <- vapply(split(res, res$Var2), function(x) sd(x$Freq, na.rm = TRUE) / sqrt(nrow(x)), FUN.VALUE = numeric(1))
+    avg <- vapply(split(res, res$Var2),
+      function(x) mean(x$Freq, na.rm = TRUE),
+      FUN.VALUE = numeric(1)
+    )
+    se <- vapply(split(res, res$Var2),
+      function(x) sd(x$Freq, na.rm = TRUE) / sqrt(nrow(x)),
+      FUN.VALUE = numeric(1)
+    )
   } else {
     db <- data.frame(id = id, group = group, par = par)
-    res <- table(db$id, db$group, db$par)
-    res <- data.frame(res)
 
-    tmp <- db |>
-      count(group, par, id) |>
-      group_by(group, par) |>
-      summarise(avg = mean(n, na.rm = TRUE), se = sd(n, na.rm = TRUE) / sqrt(n())) |>
-      tidyr::pivot_wider(id_cols = par, names_from = group, values_from = c("avg", "se"))
-
-    avg <- tmp |>
-      select(starts_with("avg")) |>
-      as.data.frame()
-    names(avg) <- u_group
-
-    se <- tmp |>
-      select(starts_with("se")) |>
-      as.data.frame()
-    names(se) <- u_group
+    # Count number of observations per group, par, and id.
```

Comment:
Certainly more verbose than `dplyr`, but can be achieved using `split`, `lapply`, and `do.call(rbind)`